admin/src/app/foms/fom-detail/fom-detail.component.ts

```diff
@@ -182,19 +182,24 @@ export class FomDetailComponent implements OnInit, OnDestroy {
   }
 
   public async publishFOM(){
-    const ready = this.validatePublishReady();
-    if (ready) {
-      this.workflowStateChangeRequest.workflowStateCode = WorkflowStateEnum.Published;
-      this.workflowStateChangeRequest.revisionCount = this.project.revisionCount;
-
-      this.isPublishing = true;
-      try {
-        await this.projectService.projectControllerStateChange(this.project.id, this.workflowStateChangeRequest).toPromise();
-      } finally {
-        this.isPublishing = false;
+    const dialogRef = this.modalSvc.openConfirmationDialog(`Publishing your FOM will make it viewable to the public once commenting opens, and you will not be able to make any edits. Do you want to proceed?`, 'Publish FOM');
```

Comment:
follow the dialog in the deleteFOM and finalizeFOM method

![image](https://github.com/bcgov/nr-fom/assets/23131735/74f33087-9814-4808-b1a4-84dc95701d01)